### ISSUE TITLE
[Docs] Correct usage of FixedNoiseGaussianLikelihood in variational models

### ISSUE THREAD

**Issue by ptonner** [cornellius-gp/gpytorch#1276]
# 📚 Documentation/Examples

I couldn't find an example of how to use the `FixedNoiseGaussianLikelihood` with variational models, so I just wanted to check that I've come up with the correct solution. The main difference from exact GP models for the fixed noise case is we are often training on mini-batches, so providing the fixed noise tensor for the whole dataset isn't going to work. The implementation already seems to support what I expected to be the solution: provide the noise tensor for each mini-batch dynamically. Again, just want to double check this is the intended solution and that I haven't missed something. Here's some example code for what I am intending to do:
                        
    from gpytorch.likelihoods import FixedNoiseGaussianLikelihood                                            
    import torch                                                                                             
    import gpytorch                                                                                          
                                                                                                            
    # setup model, data, optimization ...

    # I initialize the noise parameter as None b/c I never expect to use it                                  
    fn = FixedNoiseGaussianLikelihood(None)          

    # create the elbo
    mll = gpytorch.mlls.VariationalELBO(fn, model, num_data=N)

    # my loader is setup to provide x, y, and observation noise for each mini-batch
    for x, y, noise in loader:
        out = model(x)

        # use the noise provided for this mini-batch
        loss = -mll(out, y, noise=noise)                                                        
        loss.backward()

**Comment by ptonner** [comment]
Actually, now that I look at the `forward` method of `HomoskedasticNoise`, it looks like the dynamic noise approach in my above code would work for a standard `GaussianLikelihood` as well? I.e. if I provide noise in the `forward` call of the `mll`, this will get used to create the noise covariance. This is probably fine for my purposes, but I'm still interested if there is a preferred/correct approach.

**Comment by ptonner** [comment]
As a final note, I think my proposed approach would _not_ work if trying to use the `learn_additional_noise` option for `FixedNoiseGaussianLikelihood`, because the `noise` kwarg gets passed to both the fixed noise and the `second_noise_covar`
